src/run/troubleshoot/analysis/DockerMemoryAnalysis.ts

```diff
@@ -0,0 +1,92 @@
+import { collect, deduplicate } from "../../../helpers/array-reducers.js";
+import {
+    fetchDockerSettings,
+    DockerSettings
+} from "../../../helpers/docker-settings-store.js";
+import Service from "../../../model/Service.js";
+import { Inventory } from "../../../state/inventory.js";
+import { StateManager } from "../../../state/state-manager.js";
+import AnalysisOutcome from "./AnalysisOutcome.js";
+import { AnalysisIssue, TroubleshootAnalysisTaskContext } from "./AnalysisTask.js";
+import os from "os";
+
+const ANALYSIS_HEADLINE = "Check docker's memory allocation size";
+
+const DOCKER_MEMORY_SUGGESTIONS = [
+    "Got to settings on Docker Desktop application",
+    "Select the Resource tab - Advanced",
+    "Add to Memory Limit using the slider"
+];
+
+export default class DockerMemoryAnalysis {
+
+    static readonly ENABLED_SERVICE_COUNT_THRESHOLD:number = 10;
+    static readonly ENABLED_RESOURCE_INTENSIVE_SERVICES :string[] = ["elasticsearch"];
+
+    async analyse ({ inventory, stateManager, config }: TroubleshootAnalysisTaskContext): Promise<AnalysisOutcome> {
+        const issues = this.checkDockerMemorySize(inventory, stateManager);
+
+        return this.createOutcomeFrom(issues ? [issues] : []);
```

Comment:
If this seems to be a repeated pattern, in time you could consider creating a suprt class (perhaps Abstract) that could reuse these sort of functions. Not essential but looks familiar to the previous Analysis task you put in

Reply:
Yeah, I agree. I could work on another cleanup PR that takes care of this.